### DISCUSSION TITLE
Help per Alexa Actionable Notifications

### DISCUSSION THREAD

**Discussion by vincenzocoletta** [indomus/forum#3008] [GENERALE - Discussioni generiche]
Dopo alcuni giorni di sbattimento mi decido a chiedere aiuto agli esperti: il tema è quello in oggetto, dopo aver provato e riprovato, seguito tutti i suggerimenti nel microdettaglio e cancellato svariate volte la skill per poi ricrearla non saprei più dove sbattere la testa, mi blocco al test con il fatidico messaggio: "Si è verificato un problema con la risposta della Skill richiesta"
Credo di averle provate tutte: cambiato gli apici, effettuato l'aggiornamento alla release 0.12.2 (che poi nel lambda.function.py resta identificato come 0.12.0...), utilizzato la release con e senza binary, con SSL "True" e poi con "False" (anche se la mia connessione remota è tramite tunneling Cloudflare quindi True dovrebbe essere la configurazione corretta), Helper configurato in yaml o tramite interfaccia grafica...
Accetto ed apprezzo suggerimenti, l'unica cosa che mi viene in mente di chiedere (e perdonatemi per l'ignoranza) è se l'URL e il Token devono essere inseriti tra gli apici ed in caso affermativo quali, singoli o doppi? Non so più dove sbattere la testa...
Grazie 1000 in anticipo per l'anima pia che riuscirà a sbloccarmi...

**Comment by Scialla** [comment]
se hai fatto tutto con l'ultima versione senza binary (non dovrebbe servire con), vuol dire che invece di fare con il .git hai fatto con hello world e quindi devi fare un passo in più, ovvero sostituire il json dell'interaction model con quello che trovi qui:
https://github.com/keatontaylor/alexa-actions/blob/master/skill-package/interactionModels/custom/it-IT.json

il test poi dovrebbe funzionare

**Reply by vincenzocoletta** [reply]
Grazie Scialla! Si, avevo già provato anche con la sostituzione del json (tramite raw copy & paste da editor) in quanto già segnalato tra i commenti della guida (ormai li ho anche imparati a memoria) ma niente... Se ti viene in mente altro fammi sapere pf!
A presto e grazie,
V!

**Reply by Scialla** [reply]
Ti posso assicurare che la guida, seguita pedessiquamente, funziona.
La doc ufficiale fa scaricare lo zip in alternativa ma non specifica di sostituire il json ma, fatto questo, funziona anch'essa.

se il tuo FQDN è raggiungibile da esterno e se hai configurato la lambda function, deve funzionare

ho provato in tutti i modi e funziona sempre, l'unico problema era con la penultima versione che dava problemi e non funzionava ma, con la 12.2 (si nel codice rimane scritto 12.0) funziona

Quindi cancella la skill e riparti dalla sua creazione seguendo pedessiquamente la guida inDomus o, in alternativa, la wiki
https://github.com/keatontaylor/alexa-actions/wiki

**Reply by vincenzocoletta** [reply]
Sisi, le guide sono vangelo lo so bene; solo che non riesco a capire dove sbaglio! La generazione dell’input test mi sembra banale, spero di non aver cannato quella. Cloudflare è configurato con SSL in modalità “flessibile” (Encrypts traffic between the browser and Cloudflare) quindi direi che va bene il setting “True” nel codice della skill. Il Token l’ho anche rigenerato qualora avesse una associazione univoca con la prima skill (quella che poi ho cancellato per riprovare)… non so dove sbattere più la testa

**Reply by vincenzocoletta** [reply]
Ma per la versione “withbinary” il .json va sostituito come per quella senza binary? Mi confermi che il token va rigenerato di nuovo se la skill la cancello e riparto da zero o posso riutilizzarlo?

**Reply by Scialla** [reply]
Puoi riutilizzarlo.
Se non crei la skill usando il .git ma usando "Hello World" il json va sostituito altrimenti, durante il test, alexa in casa parla e ti dà errore

**Reply by vincenzocoletta** [reply]
Perdonami l’ignoranza ma io distinguo le due versioni con e senza binary della release 12.2 (queste per intenderci: https://github.com/keatontaylor/alexa-actions/releases/tag/v0.12.2) mentre hello world e .git non so ancora cosa siano…

**Reply by Scialla** [reply]
Si ma allora non stai usando la guida inDomus e stai usando la Wiki perché, se usassi la guida, useresti il .git e non scaricheresti gli zip (con e senza binary)

come da schermata:
![Schermata 2024-03-31 alle 10 18 52](https://github.com/indomus/forum/assets/49528981/2e595be2-46fa-4341-8bf9-6aca8ae70430)


Se invece segui la wiki ti fa scaricare lo zip e quando crei la skill non vai su import ma vai su Start from scratch (se leggi è un esempio fatto con Hello World) come da schermata:
![Schermata 2024-03-31 alle 10 21 37](https://github.com/indomus/forum/assets/49528981/f484b2c5-1181-4462-baa4-ebba90a61a5f)

sono due procedure diverse!

Riassumendo:
- o segui la guida inDomus e importi il .git e segui cosa dice la guida
oppure
- segui la wiki, non importi il .git ma vai avanti con start from scratch e successivamente importi lo zip scaricato

se segui la wiki devi poi sostituire il json con quello che ho linkato all'inizio

**Reply by vincenzocoletta** [reply]
Ok, ora ho ricollocato: io seguo la guida al 100% con il .git poi, guardando i commenti in coda alla guida, prendo i .zip e sostituisco la lambda ed infine anche il .json, corretto? La mia domanda è: il .json va sostituito sia che io prenda lo zip nobinary che quello withbinary?
Ancora grazie e scusami per l’incomprensione…

**Reply by Scialla** [reply]
se segui la guida non serve prendere gli zip e non serve sostituire il json

se segui la wiki devi sostituire il json sia con binary che senza

o segui la guida o segui la wiki, non seguirle entrambe

(i commenti in coda sono relativi al momento in cui hanno aggiornato il codice e non funzionava più, ora decidi la strada da seguire e seguine una, non mischiare perché, se non sai cosa stai facendo, rischi di incasinarti)

**Reply by vincenzocoletta** [reply]
Grazie Scialla, scusami altra piccola domanda: ma il profilo di sicurezza bisogna crearne uno nuovo oppure va bene quello già creato a suo tempo per integrare Alexa con HA seguendo questa guida: "https://indomus.it/guide/integrare-gratuitamente-amazon-echo-alexa-con-home-assistant-via-haaska-e-aws/"?

**Reply by Scialla** [reply]
Va bene quello che hai già usato

**Reply by vincenzocoletta** [reply]
Buongiorno Scialla, deve esserci ancora qualche problema sulla ultima release in quanto io sono riuscito a risolvere semplicemente utilizzando una release precedente. Ho commentato su web alla pagina della guida se può essere utile...
A presto e ancora grazie per il supporto di tutto il team!

**Reply by Scialla** [reply]
strano che tu sia riuscito con una versione precedente, quale? Perché la 12.1 aveva un bug come da [issue](https://github.com/keatontaylor/alexa-actions/issues/238) risolto poi con la 12.2.
In ogni caso, felice tu abbia risolto
